### PR TITLE
Feat/new campaign form

### LINKED ISSUE #16: Create New Campaign View
- [ ] Create a view to add a new Campaing
- [ ] Create a form component to input Campaign information
- [ ] Add New Campaign information to all Campaigns

### PR DESCRIPTION
**Pull request by edmdc**
# Pull request checklist

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Current behavior

- Campaigns lived in home folder
- No way for user to create a new campaign 

## New behavior

- Seeded campaigns live in vuex store
- New Campaign component allows user to update campaign (update to Vuex store)
- Home (All Campaigns) is updated with new campaign.   

## Other information

**Before form submission:**
image

**New Campaign form with palceholder values**
image

**Updated Campaigns page after form submission**
image

closes #16

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -10,9 +10,17 @@
...
+  name: 'App',
+  data() {
+    return {
+    }
+  },
+  computed: {
+    campaigns() {
+      return this.$store.state.campaigns
+    }
+  }
```
Comment (edmdc) [inline src/App.vue:23]: Ooops, don't quite need this at this level since each component can access store. Will delete.

File: src/store/index.js [hunk 2]
```diff
@@ -1,12 +1,39 @@
...
+        id: 4,
+        completed: false
+      }
+    ]
   },
   mutations: {
+    ADD_CAMPAIGN: (state, payload) => {
+      state.campaigns = [...state.campaigns, payload];
+    }
   },
```
Comment (edmdc) [inline src/store/index.js:36]: Allows store updates to campaigns.

File: src/components/NewCampaignForm.vue [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  },
+  methods: {
+    addCampaign() {
+      const newCampaign = {
+        title: this.title,
+        description: this.description,
+        minContribution: this.minContribution
+      }
+      this.$store.commit('ADD_CAMPAIGN', newCampaign)
+    }
```
Comment (edmdc) [inline src/components/NewCampaignForm.vue:28]: Creates campaign object and 'commits' it to the store. Use of store mutations.

File: src/views/Home.vue [hunk 4]
```diff
@@ -13,34 +13,9 @@ export default {
...
-        {
-          title: 'Help orphaned children',
-          description: 'They really need it',
-          id: 4,
-          completed: false
-        }
-      ]
+  computed: {
+    campaigns() {
+      return this.$store.state.campaigns
```
Comment (edmdc) [inline src/views/Home.vue:18]: Replaced hard coded data with global store data.

**Comment by edmdc** [comment]
Please rebase and merge over creating merge commit.

File: src/App.vue (hunk 1 above)
Comment (attom2) [reply src/App.vue:23]: You can delete it and push up again to the same branch before we merge

File: src/components/NewCampaignForm.vue (hunk 3 above)
Comment (jpc20) [reply src/components/NewCampaignForm.vue:28]: Looks great, I am just about ready to establish our connection to the blockchain. Maybe we can set this up this afternoon.

File: src/components/NewCampaignForm.vue [hunk 5]
```diff
@@ -0,0 +1,58 @@
+<template>
+  <form @submit.prevent='addCampaign'>
+    <input v-model='title' placeholder='Campaign Title' />
```
Comment (attom2) [inline src/components/NewCampaignForm.vue:3]: wow v-model is slick, much more straightforward than setting state on change

**Review by jpc20 — APPROVED** [review]
Nice work! Let's see if we can try to get the connection to the blockchain setup within the next day or so.

File: src/components/NewCampaignForm.vue (hunk 3 above)
Comment (attom2) [reply src/components/NewCampaignForm.vue:28]: How does it know to call addCampaign? I don't see that on the button click

File: src/components/NewCampaignForm.vue [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    <input v-model='minContribution' placeholder='Minimun Wei Contribution' />
```
Comment (jpc20) [inline src/components/NewCampaignForm.vue:4]: It would be great to add a conversion to $$ here.

File: src/components/NewCampaignForm.vue [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+<style lang='scss'>
+form {
+  display: flex;
+  flex-flow: column;
+  justify-content: space-around;
+  align-items: center;
+  height: 45vh;
+  width: 90vw;
+
+  input {
```
Comment (attom2) [inline src/components/NewCampaignForm.vue:42]: lovely nesting. I attempted this but I was having issues nesting a p inside of a class

File: src/views/Home.vue (hunk 4 above)
Comment (attom2) [reply src/views/Home.vue:18]: Do we use campaigns on Home.vue?

**Review by attom2 — COMMENTED** [review]
Good work on this, I had a few questions I left

**Pull request merged**